crates/codegen/src/compiler/functions.rs

```diff
@@ -126,15 +126,29 @@ impl FunctionCompiler<'_> {
 
             let param_ty = param_tys[old_idx as usize];
             if param_ty.is_aggregate() {
-                // TODO: this most likely oveshoots the abi align on some architectures
-                const ABI_ALIGN_MASK: u32 = 16 - 1;
-
-                let stack_slot = self.builder.create_sized_stack_slot(StackSlotData {
-                    kind: StackSlotKind::ExplicitSlot,
-                    size: (param_ty.stride() + ABI_ALIGN_MASK) & !ABI_ALIGN_MASK,
-                });
-
-                let stack_slot_addr = self.builder.ins().stack_addr(self.ptr_ty, stack_slot, 0);
+                // TODO: make the borrow checker happy without inlining
```

Comment:
We could do something similar to `cast`

create_stack_slot_addr could be in mod.rs and the definition here would just feed it a `&mut self.builder` and `self.ptr`

I think that's preferable to cloning the parameters, especially since this function might also be used by MemoryLoc and cast

Although fixing the A64 issue is more important than this comment